RecoLocalCalo/HcalRecAlgos/interface/HcalDeterministicFit.h

```diff
@@ -151,27 +141,22 @@ void HcalDeterministicFit::apply(const CaloSamples & cs, const std::vector<int>
         double invG = invGpar[0]+invGpar[1]*std::sqrt(2*std::log(invGpar[2]/ratio));
         float iG=0;
         getLandauFrac(-invG,-invG+tsWidth,iG);
-        ch4=(corrCharge[4]-ch3*n3)/(iG);
-        ch5=negThresh[0];
-        tsShift4=invG;
+        if (iG != 0 ) {
+	  ch4=(corrCharge[4]-ch3*n3)/(iG);
+	  ch5=negThresh[0];
```

Comment:
@mariadalfonso: Predictions that some code is temporary or will soon go away do not always come true.
The dead assignment in HcalDeterministicFit.cc is flagged by the static analyzer:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15499/14687/llvm-analysis/report-b1dc72.html#EndPath

As we mention in every Reco meeting, Reco is on a campaign to fix static analyzer warnings. We don't want to introduce new ones. I have a student working to clean up these warnings. If you won't fix them in your new code, he will do it, but that will take a new PR. It might be simpler just to fix them now in this PR.